core/Session/SaveHandler/DbTable.php

```diff
@@ -138,8 +175,25 @@ public function gc($maxlifetime)
         $sql = 'DELETE FROM ' . $this->config['name']
             . ' WHERE ' . $this->config['modifiedColumn'] . ' + ' . $this->config['lifetimeColumn'] . ' < ?';
 
-        Db::get()->query($sql, array(time()));
+        $this->query($sql, array(time()));
 
         return true;
     }
+
+    private function migrateToDbSessionTable()
+    {
+        // happens when updating from Piwik 1.4 or earlier to Matomo 3.7+
```

Comment:
Noticed this because of failing upgrade test. Tried to fix it in different ways but couldn't. Seems also quite good fix as will only be executed if table does not exist.